reactor-netty-http/src/main/java/reactor/netty/http/client/WebsocketClientOperations.java

```diff
@@ -225,7 +223,8 @@ public NettyOutbound send(Publisher<? extends ByteBuf> dataStream) {
 			return FutureMono.deferFuture(() -> {
 				if (CLOSE_SENT.getAndSet(this, 1) == 0) {
 					discard();
-					onCloseState.onNext(new WebSocketCloseStatus(frame.statusCode(), frame.reasonText()));
+					onCloseState.emitValue(new WebSocketCloseStatus(frame.statusCode(), frame.reasonText()),
+							Sinks.EmitFailureHandler.FAIL_FAST);
```

Comment:
This should probably be handled differently. Either ignore a competing attempt (which could be underlying connection closed?) or use `tryEmit` and handle return value in order to at least release the frame. Otherwise `emitValue` could raise `EmissionException` which would bubble up.

Reply:
> a competing attempt (which could be underlying connection closed?)

Isn't this guarded by `if (CLOSE_SENT.getAndSet(this, 1) == 0) {`
We should try to emit just once

Reply:
True this is only the attempt to emit. In that case `.unsafe()` with `tryEmit()` should suffice.

Reply:
please check